src/mcts/search.h

```diff
@@ -93,9 +96,19 @@ class Search {
  private:
   std::pair<Move, Move> GetBestMoveInternal() const;
 
-  // Returns a child with most visits.
-  Node* GetBestChild(Node* parent) const;
-  Node* GetBestChildWithTemperature(Node* parent, float temperature) const;
+  // Returns a random child according to (visit_fraction)^(1/temp), which
+  // simplifies to child with most visits when temp==0
+  Node* GetBestChild(Node* parent, float temperature = 0.0f) const;
+  // calculates current temperature based on ply, eval, and options passed
+  float CalculateTemperature() const;
+
+  bool NodeInSearchmoves(Node* node) const {
```

Comment:
The function is too large for header file.
Also, if it's a predicate, should start with `Is`
Also, some few words comment would be beneficial, it's not that trivial.